src/Event/EventRepository.php

```diff
@@ -45,4 +45,13 @@ public function findBySlug(string $eventSlug): ?Event
     {
         return $this->findOneBy(['slug' => $eventSlug]);
     }
+
+    /**
+     * @param string $apiSecret
+     * @return Event|Null
+     */
+    public function findByApiSecret(string $apiSecret): ?Event
+    {
+        return $this->FindOneBy(['api_secret' => $apiSecret]);
```

Comment:
metoda bez velkého písmene prosím - ono PHP to sežere, ale snažím se psát slušně case-sensitive, jako v jiných jazycích (: